### PR TITLE
Add _cat/tasks

### PR DESCRIPTION
**Pull request by imotov**
Adds new _cat endpoint that lists all tasks

### CODE REVIEW

File: core/src/main/java/org/elasticsearch/rest/action/cat/RestTasksAction.java [hunk 1]
```diff
@@ -0,0 +1,172 @@
...
+
+    @Override
+    protected void documentation(StringBuilder sb) {
+        sb.append("/_cat/tasks\n");
+    }
+
+    @Override
+    public void doRequest(final RestRequest request, final RestChannel channel, final Client client) {
+
+        boolean detailed = request.paramAsBoolean("detailed", false);
```
Comment (nik9000) [inline core/src/main/java/org/elasticsearch/rest/action/cat/RestTasksAction.java]: These feel familiar. Any chance of sharing code from RestListTasksAction?

File: rest-api-spec/src/main/resources/rest-api-spec/test/cat.tasks/10_basic.yaml [hunk 2]
```diff
@@ -0,0 +1,10 @@
+---
+"Test cat tasks output":
+
+  - do:
+      cat.tasks: {}
+
+  - match:
+      $body: |
+               / # action  task_id      parent_task_id      type    start_time   timestamp            running_time  ip                                  node
+               ^(  \S+\s+  \S+\:\d+\s+  (?:\-|\S+\:\d+)\s+  \S+\s+  \d+\s+       \d\d\:\d\d\:\d\d\s+  \S+\s+        \d{1,3}.\d{1,3}.\d{1,3}.\d{1,3}\s+  \S+(?:\s\S+)*\n)+$/
```
Comment (nik9000) [inline rest-api-spec/src/main/resources/rest-api-spec/test/cat.tasks/10_basic.yaml]: I wonder if IP is going to betray us at some point.

Comment (imotov) [inline rest-api-spec/src/main/resources/rest-api-spec/test/cat.tasks/10_basic.yaml]: This is a standard approach for all cat API calls.

File: core/src/main/java/org/elasticsearch/rest/action/cat/RestTasksAction.java (hunk 1 above)
Comment (imotov) [inline core/src/main/java/org/elasticsearch/rest/action/cat/RestTasksAction.java]: Fixed that.

File: rest-api-spec/src/main/resources/rest-api-spec/test/cat.tasks/10_basic.yaml (hunk 2 above)
Comment (nik9000) [inline rest-api-spec/src/main/resources/rest-api-spec/test/cat.tasks/10_basic.yaml]: Cool. If it bites us here it bites us everywhere. I can live with that.

**Comment by nik9000** [comment]
LGTM

**Comment by drewr** [comment]
:joy_cat: :heart:

**Pull request merged**